src/components/Article.jsx

```diff
@@ -1,15 +1,22 @@
 import { useEffect, useState } from "react"
 import { getArticle, dateFormatter } from "../utils"
-
+import { useParams } from "react-router-dom"
 
 export const Article = () => {
 
     const [result, setResult] = useState([])
     const [loading, setLoading] = useState(true)
     const [error, setError] = useState(false)
     const [errMsg, setErrMsg] = useState("Something went wrong, please navigate back to the Articles list and try again")
+    const [articleId, setArticleId] = useState(useParams().articleId)
     
-    const articleId = window.location.href.split('articles/')[1]
+   useEffect(() => {
+    setArticleId(() => {
+        return useParams().articleId
+    })
+   }, [window.location.hash])
+
```

Comment:
You don't need to use this useEffect, nor do you need articleId state - instead you can destructure articleId from useParams! 😄 It'll save you loads of processing time!

Reply:
great - thanks!  the use effect and state were to get a refresh, didn't realise use params would do this automatically